### PR TITLE
Connector-X Zones

### PR DESCRIPTION
**Pull request by having11**

### CODE REVIEW

File: src/main/include/moduledrivers/ConnectorX.h [hunk 1]
```diff
@@ -32,7 +32,6 @@ struct Configuration {
   uint16_t teamNumber;
   // Send messages to 2 other teams
   uint16_t initialTeams[2];
-  uint8_t i2c0Addr;
```
Comment (nobody5050) [inline src/main/include/moduledrivers/ConnectorX.h:35]: Why is this removed?

File: src/main/include/moduledrivers/ConnectorX.h [hunk 2]
```diff
@@ -376,20 +437,43 @@ class ConnectorXBoard : public frc2::SubsystemBase {
...
+
  private:
   Commands::Response sendCommand(Commands::Command command,
                                  bool expectResponse = false);
 
   std::unique_ptr<frc::I2C> _i2c;
   uint8_t _slaveAddress;
   LedPort _currentLedPort = LedPort::P0;
-  frc::Color8Bit _currentColors[2] = {{0, 0, 0}, {0, 0, 0}};
+  CachedDevice m_device;
```
Comment (nobody5050) [inline src/main/include/moduledrivers/ConnectorX.h:475]: Ayyyy

**Review by nobody5050 — APPROVED** [review]

File: src/main/include/moduledrivers/ConnectorX.h (hunk 1 above)
Comment (having11) [reply src/main/include/moduledrivers/ConnectorX.h:35]: address is set with DIP switches now
